### PR TITLE
Add Superparticle version of getParticlePosition.

### PR DESCRIPTION
**Pull request by atmyers**
This will allow us to more easily support RZ in parser functions, such as the one used for plotfile filtering.

### CODE REVIEW

File: Source/Particles/Pusher/GetAndSetPosition.H [hunk 1]
```diff
@@ -51,20 +52,49 @@ struct GetParticlePosition
...
+        unpack_particle(p, x, y, z);
+    }
+
+private:
+    /** \brief Extract the cartesian position coordinates of the particle
+    *         p and store them in the variables `x`, `y`, `z`
+    *         This is the implementaton and can't be called directly */
+    template <typename P>
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void unpack_particle (const P& p, RType& x, RType& y, RType& z) const noexcept
```
Comment (ax3l) [inline Source/Particles/Pusher/GetAndSetPosition.H]: Offline discussed: we move this out to a free-standing `get_particle_position` function
```C++
template <typename P>
AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
void unpack_particle_position (const P& p, const RType theta,
                               RType& x, RType& y, RType& z) const noexcept
```

File: Source/Particles/Pusher/GetAndSetPosition.H [hunk 2]
```diff
@@ -15,6 +15,29 @@
...
+    x = r*std::cos(theta);
+    y = r*std::sin(theta);
+    z = p.pos(1);
+#elif WARPX_DIM_3D
+    x = p.pos(0);
+    y = p.pos(1);
+    z = p.pos(2);
+#else
+    x = p.pos(0);
+    y = m_snan;
```
Comment (ax3l) [inline Source/Particles/Pusher/GetAndSetPosition.H]:
```suggestion
    y = std::numeric_limits<RType>::quiet_NaN();
```

File: Source/Particles/Pusher/GetAndSetPosition.H [hunk 3]
```diff
@@ -15,6 +15,29 @@
 
 #include <limits>
 
+/** \brief Extract the cartesian position coordinates of the particle
+ *         p and store them in the variables `x`, `y`, `z`
+ *         This is the implementaton and can't be called directly */
+template <typename P>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void getParticlePosition (const P& p, const RType theta,
```
Comment (ax3l) [inline Source/Particles/Pusher/GetAndSetPosition.H]: Is our convention `_` for functions?
```suggestion
void get_particle_position (const P& p, const RType theta,
```

File: Source/Particles/Pusher/GetAndSetPosition.H [hunk 4]
```diff
@@ -52,20 +76,7 @@ struct GetParticlePosition
...
-#elif WARPX_DIM_3D
-        x = m_structs[i].pos(0);
-        y = m_structs[i].pos(1);
-        z = m_structs[i].pos(2);
-#else
-        x = m_structs[i].pos(0);
-        y = m_snan;
-        z = m_structs[i].pos(1);
-#endif
+        unpack_particle_position(m_structs[i], m_theta[i], x, y, z);
```
Comment (ax3l) [inline Source/Particles/Pusher/GetAndSetPosition.H]:
```suggestion
        get_particle_position(m_structs[i], m_theta[i], x, y, z);
```

File: Source/Particles/Pusher/GetAndSetPosition.H (hunk 2 above)
Comment (atmyers) [reply Source/Particles/Pusher/GetAndSetPosition.H]: Not sure if we can use that directly in device code - let me test.

File: Source/Particles/Pusher/GetAndSetPosition.H [hunk 5]
```diff
@@ -15,6 +15,31 @@
...
+    x = r*std::cos(theta);
+    y = r*std::sin(theta);
+    z = p.pos(1);
+#elif WARPX_DIM_3D
+    x = p.pos(0);
+    y = p.pos(1);
+    z = p.pos(2);
+#else
+    x = p.pos(0);
+    y = std::numeric_limits<amrex::ParticleReal>::quiet_NaN();
```
Comment (ax3l) [inline Source/Particles/Pusher/GetAndSetPosition.H:40]: looks good for the CUDA versions we support: https://cuda.godbolt.org/z/55xj1r

File: Source/Particles/Pusher/GetAndSetPosition.H [hunk 6]
```diff
@@ -15,6 +15,31 @@
 
 #include <limits>
```
Comment (ax3l) [inline Source/Particles/Pusher/GetAndSetPosition.H:17]:
```suggestion
#include <cmath>
#include <limits>
```

**Review by ax3l — APPROVED** [review]
Thank you! :sparkles:

**Pull request merged**